hw2/src/index.js

```diff
@@ -0,0 +1,97 @@
+const express = require('express');
+const app = express();
+const Joi = require('joi');
+const bodyParser = require('body-parser');
+const port = 3000;
+
+app.use(bodyParser.json());
+
+const schema = Joi.object({
+    id: Joi.number().required(),
+    login: Joi.string().min(3).max(30).required(),
+    password: Joi.string().min(8).max(20).required(),
+    age: Joi.number().min(4).max(130).required(),
+    isDeleted: Joi.boolean().required()
+
+});
+
+const validation = (userSchema) => {
+    return (req, res, next) => {
+        const { error } = userSchema.validate(req.body);
+
+        if (error) {
+            res.status(400).json(error.message);
+        }
+        next();
+    };
+};
+
+const users = [
+];
+
+app.get('/users', (req, res) => {
+    if (users.length) {
+        const { loginSubstring = '', limit = 10 } = req.query;
+        const list = users
+            .filter(user => user.login.includes(loginSubstring))
+            .sort((a, b) => a.login.localeCompare(b.login))
+            .slice(0, limit);
+        res.send(list);
+    } else {
+        res.sendStatus(404).send('User Data is empty');
+    }
+});
+
+app.post('/user', validation(schema), (req, res) => {
+    if (Object.entries(req.body).length !== 0) {
+        users.push(req.body);
+        res.sendStatus(200);
```

Comment:
200 is good enough but often 201 is returned if something is created as a result of calling post API